### ISSUE TITLE
media browser: searchable by filters

### ISSUE THREAD

**Issue by helenvarley** [upstage-org/limelight#49]
the media browser should be searchable / sortable by the following filters:

- [ ] media item name
- [ ] media type
- [ ] uploader (player)
- [ ] stage
- [ ] tags
- [ ] date uploaded

**Comment by vickis** [comment]
it would be great to be able to search by date uploaded (month rather than
year, would be more specific)
being able to search between a given date range would be really useful (eg
around a specific show or event)

and the results appearing as the search filters are selected yes please
vicki :)

**Comment by helenvarley** [comment]
agree with vi, results appearing as the search filters are selected.
for how detailed the date should be: the full detailed information (date & time) should be accessible somewhere (as it currently is when you get the information about a media item). i don't think we need to search by such detail, but it would be good if the results of the search were then chronologically displayed, so that for example if you had uploaded several media items to try things out & then wanted to keep just the latest one, you could easily see that if the list is with most recent first.
for search filters, month & year should be sufficient.

**Comment by thongteav** [comment]
Thanks for clarifying that. One more question though, how would you like the date format to be? Right now it's something like: "24/04/18 @ 04:20 PM". Would you like to keep it that way or would you prefer some other format? Perhaps like "2018-04-24 04:20 PM" or something?

**Comment by helenvarley** [comment]
my preference is how it is now - i'm a day/month/year person :)

**Comment by helenvarley** [comment]
this is great, but maybe needs a little refinement. when i click on "Name" it sorts it alphabetically ... but not first by name, because it has sorted according to media type so it lists all the avatars, alphabetically by name, then all the backdrops, alphabetically by name. how can i get it to sort all media items alphabetically by name, regardless of what media type they are?
i tried to resent the filters but i got the "something went wrong" message & now i can't access anything ... either i broke it, or someone is restarting maybe?

**Comment by helenvarley** [comment]
(now it's working again, but i still can't work out how to have the list sort by name regardless of media type)

**Comment by rohrlach** [comment]
@thongteav do you believe it's possible for multiple filters to be applied at once?

**Comment by vickis** [comment]
is it possible to have for example a search by, alphabetical with a filter
for type and has the field all?

**Comment by thongteav** [comment]
@helenvarley I believe that it's sorting by name for each media type before this issue as well. But I can look up for a way to make them sort regardless of the type.

**Comment by thongteav** [comment]
@rohrlach yes, my local testing seems to be working for applying multiple filters. But for some reasons, the month filter doesn't work on qa. And the existing records didn't assign a default uploader.

**Comment by thongteav** [comment]
@vickis sorry, I'm not too sure what you mean by that.
